skbio/io/format/stockholm.py

```diff
@@ -0,0 +1,322 @@
+        elif line.startswith("#=GR"):
+            dna_data = _parse_stockholm_line_gr(line, dna_data)
+        elif line.startswith('#=GC'):
+            positional_metadata = _parse_stockholm_line_gc(line,
+                                                           positional_metadata)
+
+    for key in dna_data.keys():
+        # Sets blank dictionaries and lists to None instead
+        # Note: _replace is not a private function, see
+        # https://docs.python.org/2/library/collections.html#namedtuple-factory
+        # -function-for-tuples-with-named-fields
+        if not dna_data[key].metadata:
+            dna_data[key] = dna_data[key]._replace(metadata=None)
+        if not dna_data[key].pos_metadata:
+            dna_data[key] = dna_data[key]._replace(pos_metadata=None)
+        # Adds each sequence to the MSA data
+        seqs.append(constructor(dna_data[key].seq,
+                                metadata=dna_data[key].metadata,
+                                positional_metadata=(dna_data
+                                                     [key].pos_metadata)))
+
+    if not seqs:
+        raise StockholmFormatError("No data present in file.")
+    if not positional_metadata:
+        positional_metadata = None
+    # Constructs TabularMSA
+    return TabularMSA(seqs, metadata=metadata,
+                      positional_metadata=positional_metadata,
+                      index=list(dna_data.keys()))
+
+
+def _parse_stockholm_line_gf(line, metadata):
+    """Takes ``#=GF`` line and returns parsed data."""
+    line = _remove_newline(line.split(' ', 2))
+    gf_feature = line[1]
+    if gf_feature in metadata.keys():
+        metadata[gf_feature] = metadata[gf_feature] + ' ' + line[2]
+    else:
+        metadata[line[1]] = line[2]
+    return metadata
+
+
+def _parse_stockholm_line_gs(line, dna_data):
+    """Takes ``#=GS`` line and returns parsed data."""
+    line = _remove_newline(line.split(' ', 3))
+    data_seq_name = line[1]
+    if data_seq_name in dna_data.keys():
+        if not dna_data[data_seq_name].metadata:
+            dna_data[data_seq_name].metadata[line[2]] = line[3]
+    else:
+        raise StockholmFormatError("Markup line references nonexistent "
+                                   "data %r." % data_seq_name)
+    return dna_data
+
+
+def _parse_stockholm_line_gr(line, dna_data):
+    """Takes ``#=GR`` line and returns parsed data."""
+    line = _remove_newline(line.split())
+    data_seq_name = line[1]
+    gr_feature = line[2]
+    if data_seq_name in dna_data.keys():
+        if gr_feature in dna_data[data_seq_name].pos_metadata.keys():
+            raise StockholmFormatError("Found duplicate GR label %r associated"
+                                       " with data label %r" % (gr_feature,
+                                                                data_seq_name))
+        dna_data[data_seq_name].pos_metadata[gr_feature] = list(line[3])
```

Comment:
`line[3]` is pretty obvious here since everything on the left of the equals has a name.